pkg/cluster/cluster.go

```diff
@@ -725,6 +735,31 @@ func (c *Cluster) initInfrastructureRoles() error {
 	return nil
 }
 
+func (c *Cluster) mergeRoleDefinitions(roleA, roleB *spec.PgUser) (*spec.PgUser, error) {
+	// human roles always win
+	// infrastructure roles take precendence over the manifest roles
+	if roleA.Name != roleB.Name {
+		return nil, fmt.Errorf("could not merge role %#v with role %#v: role names don't match")
+	}
+	for _, origin := range []spec.RoleOrigin{spec.RoleOriginSystem, spec.RoleOriginTeamsAPI} {
+		if roleA.Origin == origin {
+			return roleA, nil
+		}
+		if roleB.Origin == origin {
+			return roleB, nil
+		}
+	}
+	// one of the roles originates from the manifest, another one from the infrastructure roles
+	if roleA.Origin == spec.RoleOriginInfrastructure {
+		roleA.Flags = roleB.Flags
```

Comment:
Does this code overwrite the flags of the infra role with the flags of the manifest role ?  In other words, do manifest roles' flags have higher priority here ?